lib/App/RecordStream/Operation/parsedate.pm

```diff
@@ -0,0 +1,121 @@
+use App::RecordStream::KeyGroups;
+
+use Time::ParseDate qw< parsedate >;
+use POSIX qw< strftime >;
+
+sub init {
+  my $this = shift;
+  my $args = shift;
+
+  $this->{'KEYS'} = App::RecordStream::KeyGroups->new;
+  $this->{'TIME_FUNC'} = sub { localtime $_[0] };   # \&CORE::localtime doesn't work until 5.16 :(
+
+  # Using a single "now" is important if we're processing a lot of relative
+  # dates so that "now" doesn't drift during processing.  time() is the
+  # default, anyway.
+  $this->{'NOW'} = time;
+
+  my $options = {
+    'key|k=s'       => sub { $this->{'KEYS'}->add_groups($_[1]) },
+    'format|f=s'    => \($this->{'FORMAT'}),
+    'dmy'           => \($this->{'UK'}),
+    'past'          => \($this->{'PAST'}),
+    'future'        => \($this->{'FUTURE'}),
+    'timezone|tz=s' => \($this->{'TIMEZONE'}),
+    'relative!'     => \($this->{'RELATIVE'}),
+    'now=i'         => \($this->{'NOW'}),
+    'localtime'     => sub { $this->{'TIME_FUNC'} = sub { localtime $_[0] } },
+    'gmtime'        => sub { $this->{'TIME_FUNC'} = sub {    gmtime $_[0] } },
+  };
+  $this->parse_options($args, $options);
+
+  die "--key is required\n"    unless $this->{'KEYS'}->has_any_group;
+  die "--format is required\n" unless defined $this->{'FORMAT'};
+}
+
+sub accept_record {
+  my $this   = shift;
+  my $record = shift;
+
+  my @values = map { $record->guess_key_from_spec($_) }
+    @{ $this->{'KEYS'}->get_keyspecs_for_record($record) };
+
+  for my $date (@values) {
+    my $epoch = parsedate(
+      $$date,
+      WHOLE           => 1,
+      VALIDATE        => 1,
+      ZONE            => $this->{'TIMEZONE'},
+      PREFER_PAST     => $this->{'PAST'},
+      PREFER_FUTURE   => $this->{'FUTURE'},
+      NO_RELATIVE     => !$this->{'RELATIVE'},
+      UK              => $this->{'UK'},
+      NOW             => $this->{'NOW'},
+    );
+    next unless $epoch;
+    $$date = strftime($this->{'FORMAT'}, $this->{'TIME_FUNC'}->($epoch))
+  }
+
+  $this->push_record($record);
+  return 1;
+}
+
+sub add_help_types {
+  my $this = shift;
+  $this->use_help_type('keyspecs');
+  $this->use_help_type('keygroups');
+  $this->use_help_type('keys');
+}
+
+sub usage {
+  my $this = shift;
+  my $options = [
+    ['key|-k <keys>',        'Datetime keys to parse and reformat; may be a key spec or key group.  Required.'],
+    ['format|-f <strftime>', 'Format string for strftime(3).  Required.'],
+    ['dmy',                  'Assume dd/mm (UK-style) instead of mm/dd (US-style)'],
+    ['past',                 'Assume ambiguous years and days of the week are in the past'],
+    ['future',               'Assume ambiguous years and days of the week are in the future'],
+    ['timezone|tz <zone>',   'Assume ambiguous datetimes are in the given timezone (defaults to the local TZ)'],
+    ['relative',             'Try to parse relative dates and times (e.g. 1 hour ago)'],
+    ['localtime',            'Output formatted datetimes in localtime (the default)'],
```

Comment:
only thing I'd say, as its a thing I've done a bunch, is it would be nice to have a preset for epoch time output. or actually might be nice to have a couple of different default outputs, like YYYY-MM-DD HH:MM:SS, epoch, a pretty one (i.e. Jan. 4th, 1990), I personally hate having to look up strftime formats, because I always forget the specific formats